src/backend/base/langflow/base/tools/component_tool.py

```diff
@@ -210,7 +210,8 @@ def get_tools(
                 args_schema = create_input_schema(tool_mode_inputs)
             else:
                 args_schema = create_input_schema(self.component.inputs)
-            name = f"{self.component.name}.{output.method}"
+
+            name = f"{self.component.name or self.component.display_name or ''}.{output.method}".strip(".")
```

Comment:
I think you should use this instead because `display_name` is not compatible and will give very different results between components that have `name` and others that don't.
```suggestion
            name = f"{self.component.name or self.component.__class__.__name__ or ''}.{output.method}".strip(".")
```